packages/algob/test/lib/tx.ts

```diff
@@ -668,3 +668,53 @@ describe("Deploy, Delete transactions test in run mode", () => {
     assert.equal(res?.deleted, false);
   });
 });
+
+describe("Update transaction test in run mode", () => {
+  useFixtureProject("stateful");
+  let deployer: Deployer;
+  let algod: AlgoOperatorDryRunImpl;
+  let deployerCfg: DeployerConfig;
+  beforeEach(async () => {
+    const env = mkEnv("network1");
+    algod = new AlgoOperatorDryRunImpl();
+    deployerCfg = new DeployerConfig(env, algod);
+    deployer = new DeployerRunMode(deployerCfg);
+    sinon.stub(algod.algodClient, "getTransactionParams")
+      .returns({ do: async () => mockSuggestedParam });
+  });
+
+  afterEach(async () => {
+    (algod.algodClient.getTransactionParams as sinon.SinonStub).restore();
+  });
+
+  it("should update in run mode", async () => {
+    let execParams: types.ExecParams = {
+      type: types.TransactionType.DeploySSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: bobAcc,
+      approvalProgram: "approval.teal",
+      clearProgram: "clear.teal",
+      localInts: 1,
+      localBytes: 1,
+      globalInts: 1,
+      globalBytes: 1,
+      payFlags: {}
+    };
+    const appInfo = await executeTransaction(deployer, execParams);
+
+    // should not be stored in checkpoint if in run mode
+    assert.isUndefined(deployer.getSSC("approval.teal", "clear.teal"));
+
+    execParams = {
+      type: types.TransactionType.UpdateSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: bobAcc,
+      appID: appInfo["application-index"],
+      newApprovalProgram: "approval.teal",
+      newClearProgram: "clear.teal",
+      payFlags: {}
+    };
+
+    await executeTransaction(deployer, execParams);
+  });
+});
```

Comment:
Can we add one/two more tests here:
+ `deploySSC` in deploy mode (ie. cp is created), then do `updateSSC` and verify checkpoint(`deployer.getApp`) is same as previous one i.e checkpoint is not updated.

+ `deploySSC` in *RUN* mode. Try to update SSC in deploy mode (not sure what happens here, is the transaction rejected?) 

What do you think?

Reply:
yes, will be good to have these.